ext/OceananigansReactantExt/TimeSteppers.jl

```diff
@@ -0,0 +1,24 @@
+module TimeSteppers
+
+using Reactant
+using Oceananigans
+
+import Oceananigans.Grids: AbstractGrid
+import Oceananigans.TimeSteppers: Clock
+
+using OceananigansReactantExt: ReactantState
+
+const ReactantGrid{FT, TX, TY, TZ} = AbstractGrid{FT, TX, TY, TZ, <:ReactantState} where {FT, TX, TY, TZ}
+
+function Clock(grid::ReactantGrid)
+    FT = Float64 # may change in the future
+    t = ConcreteRNumber(zero(FT))
+    iter = ConcreteRNumber(0)
+    stage = ConcreteRNumber(0)
+    last_Δt = ConcreteRNumber(zero(FT))
+    last_stage_Δt = ConcreteRNumber(zero(FT))
+    return Clock(; time=t, iteration=iter, stage, last_Δt, last_stage_Δt)
```

Comment:
@giordano see the implementation here

Reply:
Uhm, that's already what I had (well, what you told me to do) in https://github.com/PRONTOLab/GB-25/pull/12/files#r1953572093.  So this PR simply allows us to skip creating the clock manually?

Reply:
yes! but if there is another feature you think we should put in here I am happy to tackle that

Reply:
or if there is something we should change